assets/js/app.js

```diff
@@ -49,4 +49,34 @@ var app = angular.module("Lucere", ["ngResource", "ngRoute", "dndLists"])
         templateUrl: "/js/templates/views/student_lesson.html"
       })
       .otherwise({redirectTo: "/"});
-  }]);
\ No newline at end of file
+  }]);
+  app.run(["$rootScope", "$location", "AuthService", "$route", function($rootScope, $location, AuthService, $route) {
+    $rootScope.$on("$locationChangeStart", function(event, next, current) {
+      AuthService.currentUser(function(user) {
+        // Check if the current user is trying to visit an admin route
+        var isAdminRoute = /(admin\/)/.test(next);
+        var paramArr = next.split("/");
+        var hasPermission;
+
+        if (isAdminRoute) {
+          var libId = paramArr[6];
+          hasPermission = user.administrating.some(function(adminLib) {
```

Comment:
I'm a little concerned about checking the team ids against the library ids. The library model actually has an attribute team that refers to the team model. Is there a reason we couldn't use that?